hathor/consensus/consensus.py

```diff
@@ -75,11 +76,13 @@ def update(self, base: BaseTransaction) -> None:
         assert meta.validation.is_valid()
         try:
             self._unsafe_update(base)
-        except Exception:
-            meta.add_voided_by(self._settings.CONSENSUS_FAIL_ID)
-            assert base.storage is not None
-            base.storage.save_transaction(base, only_metadata=True)
-            raise
+        except BaseException:
+            self.log.critical(
```

Comment:
Should we mark the database as dirt and prevent the full node from restarting? I feel that this situation might happen in other cases (e.g., database writing failure), so we might generalize this to a `fail_and_exit()` function.

Reply:
Done!

Reply:
Great! Thanks!